sympy/solvers/ode/single.py

```diff
@@ -359,6 +358,137 @@ def inverse(self):
         return diffcls
 
 
+class FirstExact(SingleODESolver):
+    r"""Solves 1st order exact ordinary differential equations.
+    A 1st order differential equation is called exact if it is the total
+    differential of a function. That is, the differential equation:
+    P(x,y)dx+Q(x,y)dy=0, is exact if:
+    dP(x,y)/dy = dQ(x,y)/dx
+
+    Examples
+    ========
+    Following examples are problems:1,8 from SECTION.8 of
+    "Differential Equations With Applications And Historical Notes",
+     Second Edition, George.F.Simmons
+
+     >>> from sympy import Symbol,Function
+     >>> from sympy.solvers.ode.single import ExactODESolver,SingleODEProblem
+     >>> x=Symbol('x')
+     >>> f=Function('f')
+     >>> eq=(x+2/f(x))*f(x).diff(x)+f(x)
+     >>> prob=SingleODEProblem(eq,f(x),x)
+     >>> solver=ExactODESolver(prob)
+     >>> solver.matches()
+     True
+     >>> solver.get_general_solution()
+     [C1 + x*f(x) + 2*log(f(x))]
+     >>>
+     >>> from sympy import sin
+     >>> eq=(-sin(x/f(x))/f(x))+(x*sin(x/f(x))/f(x)**2)*f(x).diff(x)
+     >>> prob=SingleODEProblem(eq,f(x),x)
+     >>> solver=ExactODESolver(prob)
+     >>> solver.matches()
+     True
+     >>> solver.get_general_solution()
+     >>> [C1 + Integral(-sin(x/f(x))/f(x), x) + Integral(x*sin(x/f(x))/f(x)**2 - Integral(x*cos(x/f(x))/f(x)**3 + sin(x/f(x))/f(x)**2, x), f(x))]
+
+    References
+    ==========
+
+    - "Differential Equations With Applications And Historical Notes",
+       Second Edition, George.F.Simmons
+    - https://en.wikipedia.org/wiki/Exact_differential_equation
+    """
+
+    hint="1st_exact"
+    has_integral=True
+    p=None
+    q=None
+
+    def _matches(self) -> bool:
+        if self.ode_problem.order!=1:
+            self._matched=False
+            return False
```

Comment:
It isn't necessary to set `_matched` in this function

Reply:
Yeah! `matches()` sets `_matched`. Updating it.